ide/wg_Find.ml

```diff
@@ -130,22 +141,52 @@ class finder name (view : GText.view) =
     method private set_normal () =
       find_entry#misc#modify_base [`NORMAL, `NAME "white"]
 
-    method private find_from backward (starti : GText.iter) =
+    method private find_from backward ?(wrapped=false) (starti : GText.iter) =
       let found =
         if backward then self#backward_search starti
         else self#forward_search starti in
       match found with
       | None ->
         if not backward && not (starti#equal view#buffer#start_iter) then
-          self#find_from backward view#buffer#start_iter
+          self#find_from backward ~wrapped:true view#buffer#start_iter
         else if backward && not (starti#equal view#buffer#end_iter) then
-          self#find_from backward view#buffer#end_iter
+          self#find_from backward ~wrapped:true view#buffer#end_iter
         else
+          let _ = Ideutils.flash_info "String not found" in
           self#set_not_found ()
       | Some (start, stop) ->
+        let text = view#buffer#start_iter#get_text ~stop:view#buffer#end_iter in
+        let rec find_all offs accum =
+          if offs > String.length text then
+            List.rev accum
+          else try
+            let i = Str.search_forward self#regex text offs in
+            let j = Str.match_end () in
+            find_all (j + 1) (i :: accum)
+          with Not_found -> List.rev accum
```

Comment:
Is it intended to have the try-with around the recursive call?

Reply:
On any iteration of the loop, if the search fails, we return the accumulated list, without making a recursive call.  This could be expressed differently, with a conditional that chooses whether to make that call.